src/aem/download-assets.js

```diff
@@ -0,0 +1,94 @@
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import fs from 'fs';
+import path from 'path';
+import chalk from 'chalk';
+
+const CONTENT_DAM_PREFIX = '/content/dam';
+
+/**
+ * Save the given blob to a file in the download folder.
+ * @param {Blob} blob - The blob to save.
+ * @param {string} jcrPath - The JCR path of the asset.
+ * @param {string} downloadFolder - The folder to download assets to.
+ * @return {Promise<void>} A promise that resolves when the blob is saved to a file.
+ */
+async function saveBlobToFile(blob, jcrPath, downloadFolder) {
+  let assetPath = path.join(downloadFolder, jcrPath.replace(CONTENT_DAM_PREFIX, ''));
+  fs.mkdirSync(path.dirname(assetPath), { recursive: true });
+
+  const buffer = Buffer.from(await blob.arrayBuffer());
+  fs.writeFileSync(assetPath, buffer);
+}
+
+/**
+ * Download the given asset URL to the download folder.
+ * @param {string} url - The URL of the asset to download.
+ * @param {number} maxRetries - The maximum number of retries for downloading an asset.
+ * @param {number} retryDelay - The delay between retries in milliseconds.
+ * @return {Promise<Blob>} A promise that resolves with the downloaded asset.
+ */
+async function downloadAssetWithRetry(url, maxRetries = 3, retryDelay = 5000) {
+  let attempts = 0;
+  while (attempts < maxRetries) {
+    try {
+      const response = await fetch(url);
+      if (!response.ok) {
+        const msg = `Failed to fetch ${url}. Status: ${response.status}.`;
+        console.info(chalk.yellow(msg));
+        throw new Error(msg);
+      }
+      return await response.blob();
+    } catch (error) {
+      attempts++;
+      if (attempts >= maxRetries) {
+        console.error(chalk.red(`Failed to download ${url} after ${maxRetries} attempts.`));
+        throw error;
+      }
+
+      console.info(chalk.yellow(`Retrying download (${url}/${maxRetries})...`));
+      const delay = retryDelay * 2 ** (attempts - 1);
+      await new Promise((resolve) => setTimeout(resolve, delay));
+    }
+  }
+}
+
+/**
+ * Function to download assets from the asset mapping file.
+ * @param {Map<string, string>} assetMapping - The content of the asset mapping file.
+ * @param {string} downloadFolder - The folder to download assets to.
+ * @param {number} maxRetries - The maximum number of retries for downloading an asset.
+ * @param {number} retryDelay - The delay between retries in milliseconds.
+ * @return {Promise<Array<PromiseSettledResult<string>>>} A promise that resolves when all assets are downloaded.
+ * Each promise in the array will resolve with the JCR path of the downloaded asset.
+ */
+export async function downloadAssets(assetMapping, downloadFolder, maxRetries = 3,retryDelay = 5000) {
```

Comment:
```suggestion
export async function downloadAssets(assetMapping, downloadFolder, maxRetries = 3, retryDelay = 5000) {
```
1. Should the `maxRetries` also be read from the `process.env.MAX_RETRIES` (similar to how it is done in package-helper.js)?
2. Same for the `retryDelay` should be just declare a const and not have it as an input (to keep the method signatures consistent)?

Reply:
To be honest, I think no one is going to mess with those environment variables.  If it doesn't work after a specific number of tries, there's something wrong.